cns/types/codes.go

```diff
@@ -0,0 +1,82 @@
+package types
```

Comment:
`types` seems like a very generic package name. These seem like part of the API contract, is there a better home for them?

Reply:
i original called the package `response` and that stutter wasn't good either. i think it wants to be with the code in the `restserver` package, but actually putting it in that package has some potential cyclical dependency problems.

Reply:
I figured cyclical dependencies had something to do with the choice to extract this out. As a CNS client, DNC also consumes these response codes, so this change will require refactoring from that end as well. From that perspective, the `cns` package itself already contains the symbols used to work with the http api, is that an option?

Reply:
it really doesn't work in any of the existing packages, import cycles 🙃
i'm on board with the root `cns/` package holding all of the contract, but it has more than just that contract at the moment, and that causes a big mess.

Reply:
In that case, maybe call this package `api` and leave a todo to move the rest of the public contract here? I think the underlying problem really is the pervasiveness of these return codes in different layers where proper errors should be used, but that is a larger fix for another day.

Reply:
I agree and I can move it, but the whole public contract should eventually get moved _back_ to the root `cns` package, imo

Reply:
I can agree with that if the goal is to have the `cns` package be very lean and not have extraneous imports for consumers :).

Reply:
+1 on this eventually going back to `cns`. Types and interfaces should be defined in `cns` root, implementations of those interfaces should exist in subpackages that are named according to the implementation. For now an `api` package is probably fine to alleviate the cyclical imports, but usually cyclical imports means the package design is all wrong, which probably requires a much larger PR to fix.